SUPPORT.md

```diff
@@ -0,0 +1,129 @@
+releases shortly after they become `latest` (i.e: the first regular release of
+is issued).
+
+[semver]: https://semver.org
+
+### Release Lines Lifecycle Stages
+
+This Maintenance & Support Policy assigns one of the following lifecycle stages
+to each in-scope `jsii` release line in existence:
+
+1. **Current**: The latest release line is always the **Current** one. This is
+   the release under active development, receiving new features, bug fixes and
+   security updates.
+
+1. **Maintenace**: Release lines in **Maintenance** stage are no longer
+   considered in active development, and no new features will be added to these.
+   They however continue tor eceive bug fixes and security updates. Users can
+   continue to use release lines in **Maintenance** stage indefinitely, but we
+   recommend they regularly upgrade to the **Current** release lines in order to
+   keep the necessary effort low.
+
+1. **End-of-Support**: Release lines in **End-of-Support** are considered
+   defunct and are no longer maintained. They no longer receive new features,
+   bug fixes, or security updates. **End-of-Support** releases may continue to
+   be usable for a period of time, but they could be indirectly broken by
+   changes in newer releases at any time. Users are advised to migrate away from
+   **End-of-Support** release lines at the earliest convenience.
+
+### Stage Transitions
+
+```mermaid
+---
+title: Release Line Lifecycle
+---
+stateDiagram-v2
+  direction LR
+
+  EoL: End-of-Support
+
+  [*] --> Current
+  Current --> Maintenance : new release line started
+  Maintenance --> EoL : after 6 months
+  EoL --> [*]
+```
+
+Whenever a new release line is started (typically with a new `x.y.0` release,
+excluding pre-releases), it becomes **Current** and the release line it replaced
+immediately enters the **Maintenace** stage.
+
+Releases stay in the **Maintenance** stage for a minimum of 6 months before they
+reach **End-of-Support**.
+
+Once a release line has reached **End-of-Support**, it remains in this stage
+indefinitely.
+
+### Timelines & Communication
+
+The `typescript` compiler mainteners start a new release line on a quarterly
+basis, and users should expect the `jsii` compiler does the same. Users should
+expect some small code changes may be necessary when upgrading from one release
+line to another, as a result of breaking changes introduced by the `typescript`
+compiler or `jsii` itself. In either case, the release notes for the first
+release in a line includes all relevant information about the breaking changes
+and instructions for users to fix their codebase.
+
+The `jsii` compiler has a built-in mechanism to inform users that they are using
+an unsupported or soon-to-become unsupported release line. A message is emitted
+to `STDERR` when invoking the `jsii` command if the current release is:
+
+- **End-of-Support** already
+- Planned to move to **End-of-Support** within the next 30 calendar days
```

Comment:
This technically doesn't exist yet, but I have a plan that is ready to execute on.